cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -561,10 +566,58 @@ func getHash(text string) string {
 	return fmt.Sprintf("%x", h.Sum32())
 }
 
+// waitForKubernetesService waits for the "Kuberntes" master service.
+// Since the health probe on the skydns container is essentially an nslookup
+// of this service, we cannot serve any DNS records if it doesn't show up.
+func waitForKubernetesService(client *kclient.Client) (svc *kapi.Service, err error) {
+	name := fmt.Sprintf("%v/%v", kapi.NamespaceDefault, kubernetesSvcName)
+	timeout := 2 * time.Minute
+	glog.Infof("Waiting for service: %v", name)
+
+	pollErr := wait.Poll(1*time.Second, timeout, func() (bool, error) {
+		svc, err = client.Services(kapi.NamespaceDefault).Get(kubernetesSvcName)
+		if err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	if pollErr != nil {
+		return nil, fmt.Errorf("Failed to get service %v within %v: %v", name, timeout, err)
+	}
+	return svc, nil
+}
+
+// setupSignalHandlers runs a goroutine that waits on SIGINT or SIGTERM and logs it
+// before exiting.
+func setupSignalHandlers() {
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, syscall.SIGINT, syscall.SIGTERM)
+	// This program should always exit gracefully logging that it received
+	// either a SIGINT or SIGTERM. Since kube2sky is run in a container
+	// without a liveness probe as part of the kube-dns pod, it shouldn't
+	// restart unless the pod is deleted. If it restarts without logging
+	// anything it means something is seriously wrong.
+	// TODO: Remove once #22290 is fixed.
+	go func() {
+		desc := "unknown signal"
+		switch <-sigChan {
+		case syscall.SIGINT:
+			desc = "SIGINT"
+			fallthrough
```

Comment:
the fallthru ensures that this will always print SIGTERM.  I think you did not want falltru and your glog needs to be outside the switch